### ISSUE TITLE
NiceHashResponse expand

### ISSUE THREAD

**Issue by Ralle001** [edualm/SwiftyNiceHash#1]
Hello,

I know its been a while since you updated this repo, but I wanted to contribute and expand the functionality of the API. I have a problem with the Response format. Im trying to get the "...rig./{rigId}" stuff working. It essentially returns with one rig's data. 

    public struct RigsDetails: Decodable {
        
        public let name: String
        public let minerStatus: String
        
        public struct Device: Decodable {
            
            public struct EnumWithDescription: Decodable {
                public let enumName: String
                public let description: String
            }
            
            enum Devices: String {
                case name
                case status
            }
            
            public let name: String
            public let status: EnumWithDescription
            
            public struct Speed: Decodable {
                
                enum Speeds: String {
                    case algorithm
                    case title
                    case speed
                    case displaySuffix
                }
                
                public let algorithm: String
                public let title: String
                public let speed: Int
                public let displaySuffix: String
            }
            
            /*public let speeds: [Speed]*/
        }
        
        public let devices: [Device]
    }

Unfortunately when I uncomment the `public let speeds: [Speed]` line the API returns with "invalidResponse". Can you help me with that?

This should be the JSON format from the API:

    {
	name : string
	minerStatus : string
	
	devices : [
		{
		name : string
		status : {
			enumName : string
			description : string
		}
		speeds : [
			{
			algorithm : string
			title : string
			speed : number
			displaySuffix : string
			}
		]
		}
	]
	stats : [
	{
	unpaidAmount : number
	profitability : number
	}
	]
    }

**Comment by edualm** [comment]
It seems right to me, maybe NiceHash returns something they shouldn't?

I don't have a rig connected anymore so can't test the API. Could you share an example response from NiceHash that fails decoding?

**Comment by Ralle001** [comment]
I dont think so because I tested it in postman, Im gonna attach the full respond. I check the URL, the permissions for the API key and everything should work just fine. And if I just want to decode on data, like only decoding the name or minerStatus it is working but when I add the Speed then it just not going to do anything. That is why I thought I made a mistake in the decoding.

[This is the JSON](https://pastebin.com/s3m6BCLt)
